pkg/providerConfigs/detokenize.go

```diff
@@ -7,308 +7,197 @@ See the LICENSE file for more details.
+	if useCloudflareOriginIssuer {
+		tokens.CertManagerIssuerAnnotation1 = "cert-manager.io/issuer: cloudflare-origin-issuer"
+		tokens.CertManagerIssuerAnnotation2 = "cert-manager.io/issuer-kind: OriginIssuer"
+		tokens.CertManagerIssuerAnnotation3 = "cert-manager.io/issuer-group: cert-manager.k8s.cloudflare.com"
+		tokens.CertManagerIssuerAnnotation4 = "external-dns.alpha.kubernetes.io/cloudflare-proxied: \"true\""
+	} else {
+		tokens.CertManagerIssuerAnnotation1 = "cert-manager.io/cluster-issuer: \"letsencrypt-prod\""
+	}
 
-		if !!fi.IsDir() {
-			return nil
-		}
-
-		// var matched bool
-		matched, _ := filepath.Match("*", fi.Name())
-
-		if matched {
-			// ignore .git files
-			if !strings.Contains(path, "/.git/") {
-				read, err := ioutil.ReadFile(path)
-				if err != nil {
-					return err
-				}
+	if tokens.CloudProvider == "k3s" {
+		tokens.K3sEndpoint = tokens.K3sServersPrivateIps[0]
+	}
 
-				newContents := string(read)
-				newContents = strings.Replace(newContents, "<GITLAB_OWNER>", tokens.GitlabOwner, -1)
+	// The fqdn is used by metaphor/argo to choose the appropriate url for cicd operations.
+	if gitProtocol == "https" {
+		tokens.GitFqdn = fmt.Sprintf("https://%v.com/", tokens.GitProvider)
+	} else {
+		tokens.GitFqdn = fmt.Sprintf("git@%v.com:", tokens.GitProvider)
+	}
+}
 
-				err = ioutil.WriteFile(path, []byte(newContents), 0)
-				if err != nil {
-					return err
-				}
-			}
-		}
-		return nil
-	})
+// renderGoTemplating - Renders the template with the given values
+// it also includes the sprig GenericFuncMap functions listed here: https://masterminds.github.io/sprig/.
+func parseTemplate(content string) (*template.Template, error) {
+	funcs := sprig.GenericFuncMap()
+	funcs["toJson"] = toJson
+	t := template.New("gitops-template").
+		Funcs(funcs).
+		Delims(leftDelimiter, rightDelimiter)
+	return t.Parse(content)
 }
 
-// DetokenizeGithubMetaphor - Translate tokens by values on a given path
-func DetokenizeGitMetaphor(path string, tokens *MetaphorTokenValues) error {
-	err := filepath.Walk(path, detokenizeGitopsMetaphor(path, tokens))
-	if err != nil {
-		return err
+func executeTemplate(t *template.Template, writer io.Writer, tokens Tokens) error {
+	switch tmplTokens := tokens.(type) {
+	case *GitopsDirectoryValues:
+		return t.Execute(writer, tmplTokens)
+	case *MetaphorTokenValues:
+		// Handle tokens of type *MetaphorTokenValues
+		return t.Execute(writer, tmplTokens)
+	default:
+		return fmt.Errorf("invalid type for tokens: %T", tokens)
 	}
-	return nil
 }
 
-// DetokenizeDirectoryGithubMetaphor - Translate tokens by values on a directory level.
-func detokenizeGitopsMetaphor(path string, tokens *MetaphorTokenValues) filepath.WalkFunc {
-	return filepath.WalkFunc(func(path string, fi os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
+func replaceTemplateVariables(content string, tokens Tokens) string {
+	regex := regexp.MustCompile(TokenRegexPattern)
```

Comment:
To prevent paying the compilation price, the `MustCompile` could be called outside of the function once and save some CPU cycles 😄